### PR TITLE
[dotnet tool] Add ability to check if a tool is installed

### PR DESCRIPTION
**Pull request by JiaqiWang18**
Addresses #23549 
Added an optional `PACKAGE_ID` argument to `dotnet tool list` to allow checking if a specific tool was installed
If argument supplied, command returns `1` if no tool with that package id was found, return `0` otherwise.

## Usage
`dotnet tool list [<PACKAGE_ID>] [options]`
### Examples
```
PS C:\Users\Jacky\Documents\workspace\sdk> dotnet tool list -g
Package Id                             Version         Commands
-------------------------------------------------------------------------
dotnet-reportgenerator-globaltool      5.0.4           reportgenerator
microsoft.dotnet-interactive           1.0.345202      dotnet-interactive

PS C:\Users\Jacky\Documents\workspace\sdk> dotnet tool list -g dotnet-reportgenerator-globaltool
Package Id                             Version      Commands
-------------------------------------------------------------------
dotnet-reportgenerator-globaltool      5.0.4        reportgenerator

PS C:\Users\Jacky\Documents\workspace\sdk> dotnet tool list -g some.other
Package Id      Version      Commands
-------------------------------------
```

```
PS C:\Users\Jacky\Documents\workspace\exp> dotnet tool list
Package Id      Version                 Commands         Manifest
-------------------------------------------------------------------------------------------------------------------------
dotnet-ef       7.0.0-rc.1.22426.7      dotnet-ef        C:\Users\Jacky\Documents\workspace\exp\.config\dotnet-tools.json
cake.tool       2.2.0                   dotnet-cake      C:\Users\Jacky\Documents\workspace\exp\.config\dotnet-tools.json

PS C:\Users\Jacky\Documents\workspace\exp> dotnet tool list dotnet-ef
Package Id      Version                 Commands       Manifest
-----------------------------------------------------------------------------------------------------------------------
dotnet-ef       7.0.0-rc.1.22426.7      dotnet-ef      C:\Users\Jacky\Documents\workspace\exp\.config\dotnet-tools.json

PS C:\Users\Jacky\Documents\workspace\exp> dotnet tool list something.else
Package Id      Version      Commands      Manifest
---------------------------------------------------
```

## Changes
* Added `PackageIdArgument` argument to `ToolListCommandParser`
* Added additional filtering logic to `ToolListGlobalOrToolPathCommand` and `ToolListLocalCommand` in order to only display the tool matches the supplied package id
* Added testcases in `ToolListLocalCommandTests` and `ToolListGlobalOrToolPathCommandTests` to ensure command works as expected when `PACKAGE_ID` is supplied

### CODE REVIEW

**Comment by JiaqiWang18** [comment]
@baronfel @joeloff

**Review by baronfel — COMMENTED** [review]
I'm sorry for the delay on this, that's inexcusable on my part.

This looks good! A couple suggestions about the placement of parse-time logic vs execution-time logic, but this is already solid and would be a welcome contribution.

**Review by baronfel — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-tool/list/ToolListGlobalOrToolPathCommand.cs [hunk 1]
```diff
@@ -37,6 +37,13 @@ internal class ToolListGlobalOrToolPathCommand : CommandBase
         public override int Execute()
         {
             var toolPathOption = _parseResult.GetValueForOption(ToolListCommandParser.ToolPathOption);
+            var packageIdArgument = _parseResult.GetValueForArgument(ToolListCommandParser.PackageIdArgument);
+
+            PackageId? packageId = null;
+            if (!string.IsNullOrWhiteSpace(packageIdArgument))
+            {
+                packageId = new PackageId(packageIdArgument);
+            }
```
[the remark this answers is not in the document]
Comment (JiaqiWang18) [reply src/Cli/dotnet/commands/dotnet-tool/list/ToolListGlobalOrToolPathCommand.cs:46]: Thank you for the feedback!
What are some invalid PackageId values that I should check for in the delegate?

[the remark this answers is not in the document]
Comment (JiaqiWang18) [reply src/Cli/dotnet/commands/dotnet-tool/list/ToolListGlobalOrToolPathCommand.cs:46]: For reference, this is the code I have
```c#
private static PackageId? parseArgument(ArgumentResult result)
{
    string rawId = string.Join(" ", result.Tokens.ToList());
    return new PackageId(rawId);
}
```

[the remark this answers is not in the document]
Comment (baronfel) [reply src/Cli/dotnet/commands/dotnet-tool/list/ToolListGlobalOrToolPathCommand.cs:46]: After digging into it further, you can ignore me entirely - I didn't realize that PackageId was one of our abstractions with pretty lax validation rules.  This is good to go as far as I'm concerned. 👍

**Comment by baronfel** [comment]
Thank you for this @JiaqiWang18!

**Pull request merged**